receiver/purefbreceiver/factory.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package purefbreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/purefbreceiver"
+
+// This file implements Factory for Array scraper.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+// NewFactory creates a factory for Pure Storage FlashBlade receiver.
+const (
+	typeStr   = "purefb"
+	stability = component.StabilityLevelDevelopment
+)
+
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiver.WithMetrics(createMetricsReceiver, stability))
+}
+
+func createDefaultConfig() component.Config {
+	return &Config{
+		HTTPClientSettings: confighttp.HTTPClientSettings{},
+		Settings: &Settings{
+			ReloadIntervals: &ReloadIntervals{
+				Array:   15 * time.Second,
+				Clients: 300 * time.Second,
```

Comment:
300 Seconds is 5 minutes. I personally find it easier to read `5 * time.Minute`.